PEP 829: Accepted (#4937)
python/peps: peps/pep-0829.rst

``sys.path`` and execute package initialization code before control is passed
to the first line of user code.

This PEP proposes:

> * A 3 year deprecating for the processing of ``import`` lines in ``.pth``

Reviewer: ```suggestion
* A 3 year deprecation period for the processing of ``import`` lines in ``.pth``
```

>   files, after which, ``import`` lines in ``.pth`` files will be silently

Reviewer: ```suggestion
  files, after which ``import`` lines in ``.pth`` files will be silently
```

  ignored.

> * A 5 year period after which a warning will be produced for the existence of

Reviewer: Wait, they're deprecated after 3 years, and then after 5 years we produce a warning?

Author: The thinking behind this is that there's a strong desire to stop processing `import` lines asap (3 years) but not to annoy people with warnings until the last active Python that _doesn't_ support 829 is EOL (5 years).  Without this, you can't "roman ride" with both a legacy `.pth` file for older Pythons and a `.start` file for newer Pythons without warnings.

Reviewer: So the earlier deprecation is silent? I know this has been decided on already, just want to make sure it's presented clearly. Perhaps something like this:

Support for `import` lines in `.pth` files will be gradually removed:
* For the first three years (expected to be Python 3.15, 3.16, and 3.17) the behavior will remain as it is now.
* For the next two years (expected to be Python 3.18 and 3.19), `import` lines will be silently ignored.
* Afterwards (expected to be Python 3.20 and higher), a warning will be produced for the existence of `import` lines in `.pth` files.

Author: Yes, that's much clearer, thanks!  Let me push an update.

Author: PTAL

Reviewer: Thanks, approved!

  ``import`` lines in ``.pth`` files.

> * Introducing a new ``.start`` file parallel to ``.pth`` files which list

Reviewer: ```suggestion
* Introducing a new ``.start`` file parallel to ``.pth`` files which lists
```
